test/unit/eigensolver/mc/test_backtransformation.cpp

```diff
@@ -151,6 +156,90 @@ void testBacktransformationEigenv(SizeType m, SizeType n, SizeType mb, SizeType
   CHECK_MATRIX_NEAR(result, mat_c, error, error);
 }
 
+template <class T>
+void testBacktransformationEigenv(comm::CommunicatorGrid grid, SizeType m, SizeType n, SizeType mb,
+                                  SizeType nb) {
+  comm::Index2D src_rank_index(std::max(0, grid.size().rows() - 1), std::min(1, grid.size().cols() - 1));
+
+  const LocalElementSize sizeC(m, n);
+  const TileElementSize blockSizeC(mb, nb);
+  const GlobalElementSize szC = globalTestSize(sizeC);
+  const Distribution distrC(szC, blockSizeC, grid.size(), grid.rank(), src_rank_index);
+  Matrix<T, Device::CPU> mat_c(std::move(distrC));
+  dlaf::matrix::util::set_random(mat_c);
+
+  const LocalElementSize sizeV(m, m);
+  const TileElementSize blockSizeV(mb, mb);
+  const GlobalElementSize szV = globalTestSize(sizeV);
+  const Distribution distrV(szV, blockSizeV, grid.size(), grid.rank(), src_rank_index);
+  Matrix<T, Device::CPU> mat_v(std::move(distrV));
+  dlaf::matrix::util::set_random(mat_v);
+
+  const SizeType nr_reflector = std::max(static_cast<SizeType>(0), m - mb - 1);
+
+  // Copy matrices locally
+  const auto mat_c_loc = dlaf::matrix::test::allGather<T>(lapack::MatrixType::General, mat_c, grid);
```

Comment:
As said here https://github.com/eth-cscs/DLA-Future/pull/385#discussion_r742123763, it can be gathered as `const T`.

Reply:
Like: 
```suggestion
  const auto mat_c_loc = dlaf::matrix::test::allGather<const T>(lapack::MatrixType::General, mat_c, grid);
```
?
In that case, it gives an error regarding the following `larf`
```
/home/lara/Documents/DLAF/DLA-Future_eigenv_bt_distr/test/unit/eigensolver/mc/test_backtransformation.cpp:259:46:   required from ‘void BackTransformationEigenSolverTestMC_CorrectnessDistributed_Test<gtest_TypeParam_>::TestBody() [with gtest_TypeParam_ = std::complex<float>]’
/home/lara/Documents/DLAF/DLA-Future_eigenv_bt_distr/test/unit/eigensolver/mc/test_backtransformation.cpp:253:1:   required from here
/home/lara/Documents/DLAF/DLA-Future_eigenv_bt_distr/test/unit/eigensolver/mc/test_backtransformation.cpp:226:19: error: invalid conversion from ‘const std::complex<float>*’ to ‘std::complex<float>*’ [-fpermissive]
  226 |       lapack::larf(lapack::Side::Left, m - mb - j, n, mat_v_loc.ptr(v_offset), 1, tau,
      |       ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                   |
      |                   const std::complex<float>*
  227 |                    mat_c_loc.ptr(GlobalElementIndex{j + mb, 0}), mat_c_loc.ld());
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
how can I solve it? thanks!

Reply:
Ah sorry, you're right. I forgot you were actually using it for computing the result in-place, not just as reference. So, it cannot be `const T`.